code/game/machinery/machinery.dm

```diff
@@ -289,21 +288,31 @@ Class Procs:
 /obj/machinery/attackby(obj/item/W, mob/user)
 	if(obj_flags & OBJ_FLAG_SIGNALER)
 		if(istype(W, /obj/item/device/assembly/signaler))
+			if(signaler)
+				to_chat(user, SPAN_WARNING("\The [src] already has a signaler attached."))
+				return
 			var/obj/item/device/assembly/signaler/S = W
 			user.drop_from_inventory(W, src)
 			signaler = S
 			S.machine = src
 			user.visible_message("<b>[user]</b> attaches \the [S] to \the [src].", SPAN_NOTICE("You attach \the [S] to \the [src]."), range = 3)
+			log_and_message_admins("has attached a signaler to \the [src].", user, get_turf(src))
 			return
 		else if(W.iswirecutter() && signaler)
-			signaler.forceMove(get_turf(user))
-			signaler.machine = null
 			user.visible_message("<b>[user]</b> removes \the [signaler] from \the [src].", SPAN_NOTICE("You remove \the [signaler] to \the [src]."), range = 3)
-			signaler = null
+			detach_signaler(get_turf(user))
 			return
 
 	return ..()
 
+/obj/machinery/proc/detach_signaler(var/turf/detach_turf)
```

Comment:
Do we need to specify a `detach_turf` as an arg?

If there is some application for that, then the proc needs to check if detach_turf exists.

I think it could be removed and `detach_signaler` can just always drop it to `get_turf(src)` though.